R/derive_extreme_event.R

```diff
@@ -102,56 +148,290 @@
 #'     PARAM = "Worst Sleeping Problems"
 #'   )
 #' )
+#'
+#' # use different mode by event
+#' adhy <- tribble(
+#'   ~USUBJID, ~AVISITN, ~CRIT1FL,
+#'   "1",             1, "Y",
+#'   "1",             2, "Y",
+#'   "2",             1, "Y",
+#'   "2",             2, NA_character_,
+#'   "2",             3, "Y",
+#'   "2",             4, NA_character_
+#' ) %>%
+#'   mutate(
+#'     PARAMCD = "ALKPH",
+#'     PARAM = "Alkaline Phosphatase (U/L)"
+#'   )
+#'
+#' derive_extreme_event(
+#'   adhy,
+#'   by_vars = exprs(USUBJID),
+#'   events = list(
+#'     event(
+#'       condition = is.na(CRIT1FL),
+#'       set_values_to = exprs(AVALC = "N")
+#'     ),
+#'     event(
+#'       condition = CRIT1FL == "Y",
+#'       mode = "last",
+#'       set_values_to = exprs(AVALC = "Y")
+#'     )
+#'   ),
+#'   order = exprs(AVISITN),
+#'   mode = "first",
+#'   keep_source_vars = exprs(AVISITN),
+#'   set_values_to = exprs(
+#'     PARAMCD = "ALK2",
+#'     PARAM = "ALKPH <= 2 times ULN"
+#'   )
+#' )
+#'
+#' # derive confirmed best overall response (using event_joined())
+#' adsl <- tribble(
+#'   ~USUBJID, ~TRTSDTC,
```

Comment:
Would an ADSL have a TRTDTC variable in it?  I can't recall seeing one like this before?

Reply:
No, I used it only to create the ADSL dataset. I wanted to avoid calling `ymd()` for all rows in the `tribble()` call.